datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -147,10 +147,6 @@ impl OptimizerRule for PushDownProjection {
                 if !scan.projected_schema.fields().is_empty() =>
             {
                 let mut used_columns: HashSet<Column> = HashSet::new();
-                // filter expr may not exist in expr in projection.
```

Comment:
This added filters to the projection since https://github.com/apache/arrow-datafusion/pull/5188 

FYI @jackwener